tests/test_parsing.py

```diff
@@ -1084,3 +1085,60 @@ def test_bind_merge() -> None:
     assert merged_config.flag is True
     assert merged_config.nested1.value == "sun"
     assert merged_config.nested2.value == "cheese"
+
+
+class Color(Enum):
+    RED = "red"
+    BLUE = "blue"
+    GREEN = "green"
+
+
+class Number(Enum):
```

Comment:
```suggestion
class Number(IntEnum):
```
As both `int` and `Enum` are supported types, it would be useful to test that when using an `IntEnum`, the `Enum` takes precedence over the `int`.